modules/cloud-run-v2/job.tf

```diff
@@ -0,0 +1,143 @@
+        for_each = local.connector == null ? [] : [""]
+        content {
+          connector = local.connector
+          egress    = try(var.revision.vpc_access.egress, null)
+        }
+      }
+      dynamic "vpc_access" {
+        for_each = try(var.revision.vpc_access.subnet, null) == null ? [] : [""]
+        content {
+          egress = var.revision.vpc_access.egress
+          network_interfaces {
+            subnetwork = var.revision.vpc_access.subnet
+            tags       = var.revision.vpc_access.tags
+          }
+        }
+      }
+      timeout         = var.revision.timeout
+      service_account = local.service_account_email
+      dynamic "containers" {
+        for_each = var.containers
+        content {
+          name    = containers.key
+          image   = containers.value.image
+          command = containers.value.command
+          args    = containers.value.args
+          dynamic "env" {
+            for_each = coalesce(containers.value.env, tomap({}))
+            content {
+              name  = env.key
+              value = env.value
+            }
+          }
+          dynamic "env" {
+            for_each = coalesce(containers.value.env_from_key, tomap({}))
+            content {
+              name = env.key
+              value_source {
+                secret_key_ref {
+                  secret  = env.value.secret
+                  version = env.value.version
+                }
+              }
+            }
+          }
+          dynamic "resources" {
+            for_each = containers.value.resources == null ? [] : [""]
+            content {
+              limits = containers.value.resources.limits
+            }
+          }
+          dynamic "ports" {
+            for_each = coalesce(containers.value.ports, tomap({}))
+            content {
+              container_port = ports.value.container_port
+              name           = ports.value.name
+            }
+          }
+          dynamic "volume_mounts" {
+            for_each = coalesce(containers.value.volume_mounts, tomap({}))
+            content {
+              name       = volume_mounts.key
+              mount_path = volume_mounts.value
+            }
+          }
+        }
+      }
+      dynamic "volumes" {
+        for_each = var.volumes
+        content {
+          name = volumes.key
+          dynamic "secret" {
+            for_each = volumes.value.secret == null ? [] : [""]
+            content {
+              secret       = volumes.value.secret.name
+              default_mode = volumes.value.secret.default_mode
+              dynamic "items" {
+                for_each = volumes.value.secret.path == null ? [] : [""]
+                content {
+                  path    = volumes.value.secret.path
+                  version = volumes.value.secret.version
+                  mode    = volumes.value.secret.mode
+                }
+              }
+            }
+          }
+          cloud_sql_instance {
```

Comment:
Won't this need a dynamic block too to avoid non-empty plans?

Reply:
Yes, good catch, added that here too.